tests/test_profiler_unittest.py

```
@@ -0,0 +1,99 @@
+         then calling on_job_end.
+      4. Verifies that the timing file and the profile file exist and contain valid data.
+    """
+    # Create a dummy Hydra configuration with our temporary directory and a fixed job name.
+    dummy_config = DummyHydraConfig(str(tmp_path), "test_job")
+
+    # Override HydraConfig.get() so that our callback uses the dummy configuration.
+    from hydra.core.hydra_config import HydraConfig  # type: ignore
+    monkeypatch.setattr(HydraConfig, "get", lambda: dummy_config)
+
+    # Create dummy arguments for the callback methods.
+    dummy_task_function = lambda: None  # The callback doesn't use this in our implementation.
```

Ruff (0.8.2)

Do not assign a `lambda` expression, use a `def`

Rewrite `dummy_task_function` as a `def`

(E731)